src/System.Text.Json/src/System/Text/Json/Node/JsonArrayEnumerator.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+
+namespace System.Text.Json
+{
+    /// <summary>
+    ///   Supports an iteration over a JSON array.
+    /// </summary>
+    public struct JsonArrayEnumerator : IEnumerator<JsonNode>
```

Comment:
readonly?

Reply:
It isn't readonly, it modifies via MoveNext...

Reply:
The internal `IEnumerator<JsonNode>` is modified, but the reference to that is always readonly, no?

Reply:
That's true for the current implementation; but if the enumerator were lifted more (including correcting the field type to be the List<T>.Enumerator struct-enumerator) then mutable-struct semantics would apply.

Since the purpose of the enumerator is to modify I think the readonly would be a bad signal.  Pass-through properties for reference types clearly make things a little murky, but I would argue against putting `readonly` as a modifier on the MoveNext method since it makes the struct now represent a different state; and (IIRC) the readonly modifier on the type just means "apply it to all the members".